oaknational/Oak-Web-Application
# feat: preview mode enhancements

## Description
- Entering preview mode should now work regardless of if there's a draft entry version of an entry or not, falling back to the published version if not.
- Lists of items like the blogs (`/blog` and homepage) will now filter out invalid items when in preview mode
- Added some documentation (`node-lib/sanity-graphql/sanity-graphql.md`) on how queries are shaped and how fixtures are generated
- Fix a bug where preview mode would error if you tried to ender it while on the homepage (thanks for making `req.query.path` undefined next, that totally makes sense)

Most of the `.gql` file changes are duplicates so they can be safely skimmed once the first is checked

The next step this unlocks is calling `.deepPartial` on schemas and returning the potential errors to the frontend to be nicely rendered for school support (in preview mode), rather than throwing a 500 and showing an unhelpful error

note: I've set `SANITY_DATASET="feat-preview-content"` in the vercel config for this branch specifically so you can verify that it's showing draft/non draft content. I've edited very page (and 2 blog posts) in that dataset to add "draft" to the title

## Issue(s)

Fixes #425
Fixes #314 

## How to test
Note: instead of manually going to the URL below, you can create a bookmarklet from the [instructions within the CMs](https://oak.sanity.studio/desk/cmsHelp;homepage) to toggle preview mode on any version of OWA

1. Go to https://oak-web-application-git-fix-preview-mode-oak-national-academy.vercel.app/api/preview?secret=LcLt4fj87uUitpZJ
2. Click around the site, ideally on each page 
3. You should see most pages have `draft` in the page title/header
4. Click `exit preview` in the corner and re-visit each of the pages, verify there's no `draft` in the title

**To go the extra mile:**
Set your local sanity studio dataset to `feat-preview-content` and fiddle with the content, verifying it works with and without draft versions of entries

## Checklist
